lib/src/cmaf/ham/mapper/hls/mapHlsToHam.ts

```diff
@@ -18,21 +18,60 @@ function mapHlsToHam(manifest: Manifest) {
 	const playlists: PlayList[] = mainManifestParsed.playlists;
 	const mediaGroupsAudio = mainManifestParsed.mediaGroups?.AUDIO;
 	const mediaGroupsSubtitles = mainManifestParsed.mediaGroups?.SUBTITLES;
-	const audioSwitchingSets: SwitchingSet[] = [];
-	const selectionSets: SelectionSet[] = [];
 	const manifestPlaylists = manifest.ancillaryManifests
```

Comment:
Maybe manifestPlaylists should be asigned like this : 
`const manifestPlaylists = manifest.ancillaryManifests ?? []`

Is it necessary to create a  copy by using [...manifest.ancillaryManifests]?

Reply:
Yes, I'm cloning the array to don't modify the original one. If you see inside the new functions that I added, I'm using `shift` method, which returns the first element and remove it from the array.

Reply:
Got it!